In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf
import matplotlib.pyplot as plt
import seaborn as sns

## Cross Sectional Data

In [2]:
df = pd.read_excel("../Econometrics4e_Replication_Files/Chapter 1/table11.xlsx")
df.head()

F:\miniconda3\envs\py\Lib\site-packages\openpyxl\worksheet\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)


Observation Number  Average Test Score  Student-Teacher Ratio  \
0                   1          690.799988              17.889910   
1                   2          661.200012              21.524664   
2                   3          643.599976              18.697226   
3                   4          647.700012              17.357143   
4                   5          640.849976              18.671329   

   Expenditure per Pupil  % of Students Learning English  
0            6384.911133                        0.000000  
1            5099.380859                        4.583333  
2            5501.954590                       30.000002  
3            7101.831055                        0.000000  
4            5235.987793                       13.857677

the data on test scores in California school districts are cross sectional. Those data are for 420
entities (school districts) for a single time period (1999)

In [3]:
df.shape

(420, 5)

In [4]:
# Average score for first district
df["Average Test Score"].iloc[0]

np.float64(690.7999877929688)

In [5]:
# first row, second column (same)
df.iloc[0,1]

np.float64(690.7999877929688)

In [6]:
# Average student teacher ratio for first district
df["Student-Teacher Ratio"].iloc[0]

np.float64(17.889909744262695)

In [7]:
# Average expenditure per pupil in district 1 
df["Expenditure per Pupil"].iloc[0]

np.float64(6384.9111328125)

In [8]:
# % of Students Learning English
df["% of Students Learning English"].iloc[0]

np.float64(0.0)

## Time Series Data

In [9]:
df2 = pd.read_excel("../Econometrics4e_Replication_Files/Chapter 1/table12.xlsx")
df2.head()

F:\miniconda3\envs\py\Lib\site-packages\openpyxl\worksheet\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)


Observation Number Date (Year:Quarter)  GDP Growth Rate  Term Spread
0                   1            1960:Q1          8.812926     0.613333
1                   2            1960:Q2         -1.520836     1.266667
2                   3            1960:Q3          0.994741     1.473333
3                   4            1960:Q4         -4.871651     1.580000
4                   5            1961:Q1          2.710556     1.436667

In [10]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 232 entries, 0 to 231
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Observation Number   232 non-null    int64  
 1   Date (Year:Quarter)  232 non-null    object 
 2   GDP Growth Rate      232 non-null    float64
 3   Term Spread          232 non-null    float64
dtypes: float64(2), int64(1), object(1)
memory usage: 7.4+ KB


In [11]:
def convert_quarter_to_date(quarter_str):
    year, quarter = quarter_str.split(':')
    month = (int(quarter[1]) - 1) * 3 + 1  
    return pd.Timestamp(year=int(year), month=month, day=1)

In [12]:
df2['Date'] = df2['Date (Year:Quarter)'].apply(convert_quarter_to_date)

In [13]:
df2.head()

Observation Number Date (Year:Quarter)  GDP Growth Rate  Term Spread  \
0                   1            1960:Q1          8.812926     0.613333   
1                   2            1960:Q2         -1.520836     1.266667   
2                   3            1960:Q3          0.994741     1.473333   
3                   4            1960:Q4         -4.871651     1.580000   
4                   5            1961:Q1          2.710556     1.436667   

        Date  
0 1960-01-01  
1 1960-04-01  
2 1960-07-01  
3 1960-10-01  
4 1961-01-01

In [14]:
# starting GDP
df2.iloc[0,2]

np.float64(8.81292622284)

In [15]:
df2.shape

(232, 5)

 Term speed of first observation is calculated as  
long term interest - short interest = 4.5 - 3.9 = 0.6

## Panel Data

In [16]:
df3 = pd.read_excel("../Econometrics4e_Replication_Files/Chapter 1/table13.xlsx")
df3.head()

F:\miniconda3\envs\py\Lib\site-packages\openpyxl\worksheet\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)


Observation Number State  Year  Packs per Capita  Avg. Price  Total Taxes
0                   1    AL  1985        116.486282    1.021817     0.333483
1                   2    AR  1985        128.534592    1.014750     0.370000
2                   3    AZ  1985        104.522614    1.085788     0.361704
3                   4    CA  1985        100.363037    1.078373     0.321040
4                   5    CO  1985        112.963539    0.942667     0.310000

In [17]:
df3.shape

(528, 6)

In the cigarette data set, we have observations on n = 48 continental U.S. states (entities) for T = 11 years (time periods) from
1985 to 1995. Thus, there is a total of n * T = 48 * 11 = 528 observations.

In [18]:
df3.iloc[[48]]

Observation Number State  Year  Packs per Capita  Avg. Price  Total Taxes
48                  49    AL  1986        117.159317    1.079892     0.334058

First 48 observations cover year 1985

In [19]:
df3[df3['State'] == 'AL']

Observation Number State  Year  Packs per Capita  Avg. Price  Total Taxes
0                     1    AL  1985        116.486282    1.021817     0.333483
48                   49    AL  1986        117.159317    1.079892     0.334058
96                   97    AL  1987        115.836655    1.135273     0.334607
144                 145    AL  1988        115.258446    1.200334     0.335251
192                 193    AL  1989        109.205956    1.332560     0.336560
240                 241    AL  1990        111.744873    1.434486     0.337569
288                 289    AL  1991        107.014748    1.617212     0.359378
336                 337    AL  1992        106.902931    1.761137     0.380803
384                 385    AL  1993        104.957794    1.647343     0.399677
432                 433    AL  1994        103.426323    1.562756     0.418839
480                 481    AL  1995        101.085434    1.583713     0.419047

In [20]:
AL = df3.index[df3['State'] == 'AL'].tolist()
AL

[0, 48, 96, 144, 192, 240, 288, 336, 384, 432, 480]

In [21]:
df3.loc[AL]

Observation Number State  Year  Packs per Capita  Avg. Price  Total Taxes
0                     1    AL  1985        116.486282    1.021817     0.333483
48                   49    AL  1986        117.159317    1.079892     0.334058
96                   97    AL  1987        115.836655    1.135273     0.334607
144                 145    AL  1988        115.258446    1.200334     0.335251
192                 193    AL  1989        109.205956    1.332560     0.336560
240                 241    AL  1990        111.744873    1.434486     0.337569
288                 289    AL  1991        107.014748    1.617212     0.359378
336                 337    AL  1992        106.902931    1.761137     0.380803
384                 385    AL  1993        104.957794    1.647343     0.399677
432                 433    AL  1994        103.426323    1.562756     0.418839
480                 481    AL  1995        101.085434    1.583713     0.419047

## Key Concept

• Cross-sectional data consist of multiple entities observed at a single time
period.  
• Time series data consist of a single entity observed at multiple time periods.  
• Panel data (also known as longitudinal data) consist of multiple entities,
where each entity is observed at two or more time periods.  